docs/guides/server/logging.adoc

```diff
@@ -176,6 +176,53 @@ If you want to have JSON logs in *ECS* (Elastic Common Schema) format for the co
 {"@timestamp":"2025-02-03T14:53:22.539484211+01:00","event.sequence":9608,"log.logger":"io.quarkus","log.level":"INFO","message":"Keycloak 999.0.0-SNAPSHOT on JVM (powered by Quarkus 3.17.8) started in 4.615s. Listening on: http://0.0.0.0:8080","process.thread.name":"main","process.thread.id":1,"mdc":{},"ndc":"","host.hostname":"host-name","process.name":"/usr/lib/jvm/jdk-21.0.3+9/bin/java","process.pid":77561,"data_stream.type":"logs","ecs.version":"1.12.2","service.environment":"prod","service.name":"Keycloak","service.version":"999.0.0-SNAPSHOT"}
 ----
 
+=== Asynchronous logging
+{project_name} supports asynchronous logging, which might be useful for deployments requiring **high throughput** and **low latency**.
+Asynchronous logging uses a separate thread to take care of processing all log records.
+The logging handlers are invoked in exactly the same way as with synchronous logging, only done in a separate thread.
+You can enable asynchronous logging for all {project_name} log handlers.
+
+The underlying mechanism for asynchronous logging uses a queue for processing log records.
+Every new log record is added to the queue and then published to every log handler with enabled asynchronous logging.
+
+If the queue is already full, it either blocks until it is available or discards the log record, depending on the configuration.
+
+NOTE: Be aware that enabling asynchronous logging might bring some **additional computational overhead** due to the additional separate thread.
```

Comment:
Wouldn't memory be more of a concern than cpu? There is the additional stack space for the thread, and the additional heap space for the queue.

The actual cpu overhead of context switching should be very low.

Reply:
@shawkins Nice point, agree.